[WebDriver] Address unchecked optional accesses reported by clang-tidy

lauromoura opened the pull request:
#### 9b483097e475a1ea81b40c46f95bedb08f79dd06
<pre>
[WebDriver] Address unchecked optional accesses reported by clang-tidy
<a href="https://bugs.webkit.org/show_bug.cgi?id=295617">https://bugs.webkit.org/show_bug.cgi?id=295617</a>

Reviewed by BJ Burg.

clang-tidy `bugprone-unchecked-optional-access` reported a number of
cases where we were accessing the.optional values directly without
checks.

This commit also shields the m_toplevelBrowsingContext value access
behind an &quot;unchecked...&quot; method.

* Source/WebDriver/CommandResult.cpp:
* Source/WebDriver/CommandResult.h:
* Source/WebDriver/Session.cpp:
* Source/WebDriver/Session.h:

Canonical link: <a href="https://commits.webkit.org/297461@main">https://commits.webkit.org/297461@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5cfb1caa37e9e9fff8b32343305e97eb1e53cd7f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/111529 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/31195 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/21660 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/117561 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/61800 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/31876 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/39777 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/84744 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/35534 "") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/114476 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/25444 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/100382 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/65192 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/24791 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/18518 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/61387 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/94828 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/18583 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/120798 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/38578 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/28659 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/93669 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/38954 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/96650 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/93492 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23898 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/38611 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/16389 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/34618 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/38467 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/43944 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/38134 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/41465 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/39834 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

lauromoura commented on Source/WebDriver/Session.cpp:
```diff
@@ -214,6 +215,7 @@ void Session::switchToBrowsingContext(const String& browsingContext, Function<vo
     }
 
     auto parameters = JSON::Object::create();
+    ASSERT(m_toplevelBrowsingContext);
```
Went with `ASSERT`, but maybe should we use `RELEASE_ASSERT` instead?

burg commented on Source/WebDriver/Session.cpp:
```diff
@@ -263,6 +266,7 @@ void Session::createTopLevelBrowsingContext(Function<void(CommandResult&&)>&& co
 void Session::handleUserPrompts(Function<void(CommandResult&&)>&& completionHandler)
 {
     auto parameters = JSON::Object::create();
+    ASSERT(m_toplevelBrowsingContext);
```
I'd prefer we put this in a getter that asserts, like `uncheckedTopLevelBrowsingContext()`

lauromoura committed:
[WebDriver] Address unchecked optional accesses reported by clang-tidy
https://bugs.webkit.org/show_bug.cgi?id=295617

Reviewed by BJ Burg.

clang-tidy `bugprone-unchecked-optional-access` reported a number of
cases where we were accessing the.optional values directly without
checks.

This commit also shields the m_toplevelBrowsingContext value access
behind an "unchecked..." method.

* Source/WebDriver/CommandResult.cpp:
* Source/WebDriver/CommandResult.h:
* Source/WebDriver/Session.cpp:
* Source/WebDriver/Session.h:

Canonical link: https://commits.webkit.org/297461@main
